### PR TITLE
Update aggregator.py

### PR DESCRIPTION
**Pull request by zina-cs**
### Changes

Added an error message

### Reason for changes

send warning message to avoid Inconsistencies arise when the dataset size is less than the provided or default 'subset_size'.

### Related tickets

Closes: https://github.com/openvinotoolkit/nncf/issues/2562

I had an inquiry:
I noticed that subset_size is sometimes put as 100, or 300, or specified in the advanced parameters. Should a default be used here, or could you point me to where I can find the correct subset_size to be imported?

### CODE REVIEW

File: nncf/common/tensor_statistics/aggregator.py [hunk 1]
```diff
@@ -82,6 +84,11 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
             empty_statistics = False
         if empty_statistics:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        
+        if len(self.dataset) < subset_size:
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:88]: There is no guarantee that `__len__` is implemented for the dataset. In the `Statistics collection` loop, you can count the number of processed input_data, and after the loop, you can check if `subset_size` is higher than the number of samples in the provided dataset. I think we should display a warning instead of an error in this case.

File: nncf/common/tensor_statistics/aggregator.py [hunk 2]
```diff
@@ -79,9 +85,13 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
...
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+        
+        if processed_samples == 0:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        
+        if subset_size > processed_samples:
+            warnings.warn(f"Dataset contains only {processed_samples} samples, smaller than the requested subset size {subset_size}.")
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:94]:
```suggestion
            nncf_logger.warning(f"Dataset contains only {processed_samples} samples, smaller than the requested subset size {subset_size}.")  
```

File: nncf/common/tensor_statistics/aggregator.py [hunk 3]
```diff
@@ -32,6 +34,8 @@
 )
 
 
+
+
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:38]: Please avoid making unnecessary changes

File: nncf/common/tensor_statistics/aggregator.py [hunk 4]
```diff
@@ -79,9 +82,13 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
...
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+        
+        if processed_samples == 0:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        
+        if subset_size > processed_samples:
+            nncf_logger.warning(f"Dataset contains only {processed_samples} samples, smaller than the requested subset size {subset_size}.")
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:91]:
```suggestion
            nncf_logger.warning(
                f"Dataset contains only {processed_samples} samples, smaller than the requested subset size {self.stat_subset_size}."
            )
```

**Comment by zina-cs** [comment]
Great, any other changes required to close this issue?

**Comment by l-bat** [comment]
> Great, any other changes required to close this issue?

No, once the PR passes all checks, we’ll merge it

File: nncf/common/tensor_statistics/aggregator.py [hunk 5]
```diff
@@ -79,9 +82,13 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
...
             processed_outputs = self._process_outputs(outputs)
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+        
+        if processed_samples == 0:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        
+        if subset_size > processed_samples:
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:90]:
```suggestion
        if self.stat_subset_size > processed_samples:
```

File: nncf/common/tensor_statistics/aggregator.py [hunk 6]
```diff
@@ -23,6 +23,7 @@
 from nncf.data.dataset import DataItem
 from nncf.data.dataset import Dataset
 from nncf.data.dataset import ModelInput
+from nncf.common.logging import nncf_logger
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:26]: linter: move this line after
 `from nncf.common.graph.transformations.layout import TransformationLayout`

File: nncf/common/tensor_statistics/aggregator.py [hunk 7]
```diff
@@ -70,7 +71,9 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
         engine = factory.EngineFactory.create(model_with_outputs)
 
         iterations_number = self._get_iterations_number()
-        empty_statistics = True
+
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:74]: linter: remove trailing whitespaces

File: nncf/common/tensor_statistics/aggregator.py [hunk 8]
```diff
@@ -79,9 +82,16 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
             outputs = engine.infer(input_data)
             processed_outputs = self._process_outputs(outputs)
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:86]: linter: remove trailing whitespaces

File: nncf/common/tensor_statistics/aggregator.py [hunk 9]
```diff
@@ -79,9 +82,16 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
             outputs = engine.infer(input_data)
             processed_outputs = self._process_outputs(outputs)
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+        
+        if processed_samples == 0:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+
```
Comment (l-bat) [inline nncf/common/tensor_statistics/aggregator.py:89]: linter: remove trailing whitespaces

**Review by l-bat — APPROVED** [review]
Thank you!

**Comment by zina-cs** [comment]
Thank you for your assistance!!

File: nncf/common/tensor_statistics/aggregator.py [hunk 10]
```diff
@@ -79,9 +79,14 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
...
             processed_outputs = self._process_outputs(outputs)
             self._register_statistics(processed_outputs, merged_statistics)
-            empty_statistics = False
-        if empty_statistics:
+            processed_samples += 1
+        if processed_samples == 0:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        if self.stat_subset_size is not None and self.stat_subset_size > processed_samples:
+            nncf_logger.warning(
+                f"Dataset contains only {processed_samples} samples, "
```
Comment (alexsu52) [inline nncf/common/tensor_statistics/aggregator.py:87]: @l-bat, @eaidova, @nikita-savelyevv please check that this is not impact NNCF examples, OpenVINO notebooks and optimum-intel.

Comment (l-bat) [reply nncf/common/tensor_statistics/aggregator.py:87]: @zina-cs Could you please provide `subset_size` to the `nncf.quantize` in the following examples?
1. https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/openvino: 
    * anomaly_stfpm_quantize_with_accuracy_control - `subset_size=109`
    * yolov8 - `subset_size=128`
    * yolov8_quantize_with_accuracy_control - `subset_size=128`
2. https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/onnx/yolov8_quantize_with_accuracy_control - `subset_size=128`
3. https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/torch/ssd300_vgg16 - `subset_size=128`
